caffe2/contrib/tensorrt/tensorrt_tranformer.cc

```diff
@@ -373,32 +285,58 @@ OperatorDef TensorRTTransformer::BuildTrtOp(
   return op;
 }
 
-void TensorRTTransformer::ClusterToTrtOp(
+NetDef TensorRTTransformer::SubnetToTrtOp(
+    const caffe2::NetDef& net,
     Workspace* ws,
-    const NetDef& pred_net,
-    int start,
-    int end,
-    const std::unordered_set<std::string>& weights,
-    const std::unordered_map<std::string, TensorShape>& shape_hints,
-    ::ONNX_NAMESPACE::ModelProto* model,
-    std::vector<OperatorDef>* new_ops) {
-  if (model->graph().node_size() == 0) {
-    return;
+    onnx::OnnxExporter* exporter,
+    std::unordered_map<std::string, TensorShape>* shape_hints) {
+  ::ONNX_NAMESPACE::ModelProto onnx_model;
+  FillModelInfo(&onnx_model);
+
+  // Convert c2 ops to onnx ops, add const weights if there are any
+  for (const auto& op : net.op()) {
+    const auto results = exporter->Caffe2OpToOnnxNodes(op, *shape_hints);
+    const auto& node_protos = results.first;
+    for (const auto& n : node_protos) {
+      onnx_model.mutable_graph()->add_node()->CopyFrom(n);
+    }
+    for (const auto& t : results.second) {
+      VLOG(2) << "Adding extra init tensor: " << t.name();
+      TensorShape shape;
+      shape.mutable_dims()->CopyFrom(t.dims());
+      shape_hints->emplace(t.name(), std::move(shape));
+      ::ONNX_NAMESPACE::TensorProto tf;
+      tf.set_name(t.name());
+      tf.mutable_dims()->CopyFrom(t.dims());
+      tf.set_data_type(::ONNX_NAMESPACE::TensorProto::FLOAT);
+      std::vector<int64_t> v;
+      v.resize(t.raw_data().size() / sizeof(int64_t));
```

Comment:
Hmm, I went sloppy on that, will fix.

Reply:
Actually, this is because that the onnx model is generated by `Caffe2OpToOnnxNodes`, which will always generate raw tensor. 
https://github.com/pytorch/pytorch/blob/6252706feb23e448bff944c3505092b18718d3ab/caffe2/onnx/onnx_exporter.cc#L77